kura/org.eclipse.kura.linux.bluetooth/src/main/java/org/eclipse/kura/linux/bluetooth/le/beacon/BluetoothAdvertisingData.java

```diff
@@ -20,6 +20,14 @@
     private static final String MANUFACTURER_AD = "ff";
     private static final String BEACON_ID = "0215";
 
+    private static final int MAJOR_MAX = 65535;
+    private static final int MAJOR_MIN = 0;
+    private static final int MINOR_MAX = 65535;
+    private static final int MINOR_MIN = 0;
+    private static final short TX_POWER_MAX = 126;
+    private static final short TX_POWER_MIN = -127;
+    private static final String UUID_DEFAULT = "aaaaaaaabbbbccccddddeeeeeeeeeeee";
+
     public static String getData(String uuid, Integer major, Integer minor, String companyCode, Integer txPower,
```

Comment:
The checks for major, minor, txpower and uuid have been already performed at application level (i.e. in the BeaconExample), so they are run twice...

Reply:
It's actually better to do these checks, here (too). If someone implements their own advertiser and doesn't add the checks, Kura should still take care that the checks are performed and invalid values are not used.

Reply:
👍